lib/iex/lib/iex/helpers.ex

```diff
@@ -631,6 +663,68 @@ defmodule IEx.Helpers do
     IO.puts("#{pad_key(key)}#{format_bytes(value)}")
   end
 
+  defp print_allocator_sizes(%{
+         block_size: block_size,
+         carrier_size: carriers_size,
+         max_carrier_size: max_carriers_size
+       }) do
+    IO.puts("#{pad_key("Block size")}#{format_bytes(block_size, :KB)}")
+    IO.puts("#{pad_key("Carrier size")}#{format_bytes(carriers_size, :KB)}")
+    IO.puts("#{pad_key("Max Carrier size")}#{format_bytes(max_carriers_size, :KB)}")
+  end
+
+  defp get_allocator_sizes() do
+    alloc_util_allocators = :erlang.system_info(:alloc_util_allocators)
+
+    allocators_map =
+      Map.new(
+        [:total | alloc_util_allocators],
+        &{&1, %{block_size: 0, carrier_size: 0, max_carrier_size: 0}}
+      )
+
+    try do
+      for {allocator, allocator_instances} <-
+            :erlang.system_info({:allocator_sizes, alloc_util_allocators}),
+          {:instance, _, instance_info} <- allocator_instances,
+          {_, [{:blocks, blocks}, {:carriers_size, ccs, mcs, _}]} <-
+            instance_info,
+          reduce: allocators_map do
+        acc ->
+          cbs =
+            case blocks do
+              [{_, [{:size, bs, _, _}]}] -> bs
+              _ -> 0
+            end
+
+          %{
+            block_size: block_size,
+            carrier_size: carrier_size,
+            max_carrier_size: max_carrier_size
+          } = Map.get(acc, allocator)
+
+          %{
+            block_size: total_block_size,
+            carrier_size: total_carrier_size,
+            max_carrier_size: total_max_carrier_size
+          } = Map.get(acc, :total)
+
+          acc
+          |> Map.replace(allocator, %{
+            block_size: block_size + cbs,
+            carrier_size: carrier_size + ccs,
+            max_carrier_size: max_carrier_size + mcs
+          })
+          |> Map.replace(:total, %{
+            block_size: total_block_size + cbs,
+            carrier_size: total_carrier_size + ccs,
+            max_carrier_size: total_max_carrier_size + mcs
+          })
+      end
+    rescue
+      _ -> allocators_map
```

Comment:
Shouldn't we return an empty instead of a map where everything is 0?

Reply:
yes